database/engine/metadata_log/metadatalogapi.c

```diff
@@ -3,6 +3,21 @@
 
 #include "metadatalog.h"
 
+static inline struct metalog_instance *get_metalog_ctx(RRDHOST *host)
+{
+    if (!host->rrdeng_ctx) {
+        if (host != localhost && !localhost->rrdeng_ctx)
+            return NULL;
+        else {
+            if (localhost->rrdeng_ctx)
```

Comment:
Is it possible for `localhost` to be `NULL` at this point?

Reply:
no it would have failed way before that